### ISSUE TITLE
When a boolean parameter to a cypher directive is false NULL is inserted instead

### ISSUE THREAD

**Issue by michaeldaudiolo** [neo4j/graphql#4831]
**Describe the bug**
It seems like this bug was actually introduced in 4.2.0. When a boolean field argument is used in a `@cypher` directive, if the boolean is false, instead of the parameter being inserted into the cypher query NULL is inserted instead.

**Type definitions**
```
type Test {
    test(value: Boolean): Boolean! @cypher(statement: "RETURN $value as value", columnName: "value")
}
```

**To Reproduce**

1. With the type definition above execute the following mutation to create a Test instance
```
mutation Mutation($input: [TestCreateInput!]!) {
  createTests(input: $input) {
    tests {
      test
    }
  }
}
```
variables:
```
{
  "input": [
    {
      "_emptyInput": true
    }
  ]
}
```

2. Then run the following query
```
query ExampleQuery {
  tests {
    test(value: true)
  }
}

```

3. See return value:
```
{
  "data": {
    "tests": [
      {
        "test": true
      }
    ]
  }
}
```
and generated cypher:
```
MATCH (this:Test)
CALL {
    WITH this
    CALL {
        WITH this
        WITH this AS this
        RETURN $param0 as value
    }
    UNWIND value AS this0
    RETURN head(collect(this0)) AS this0
}
RETURN this { test: this0 } AS this
```

4. Run the same query with false instead
```
query ExampleQuery {
  tests {
    test(value: false)
  }
}

```

5. See error: `Cannot return null for non-nullable field Test.test.`
and generated cypher:
```
MATCH (this:Test)
CALL {
    WITH this
    CALL {
        WITH this
        WITH this AS this
        RETURN NULL as value
    }
    UNWIND value AS this0
    RETURN head(collect(this0)) AS this0
}
RETURN this { test: this0 } AS this
```

**Expected behavior**
Boolean false values are used in the cypher statement properly, and no errors occur.

**Additional context**
This issue seems to have been introduced in 4.2.0. If you use a default value for the field argument like `test(value: Boolean = false): Boolean! @cypher(...)` and don't use any variables in the query it behaves the same and also errors.

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @michaeldaudiolo. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by mjfwebb** [comment]
Hi @michaeldaudiolo! I'm running through your reproduction steps and I received an error on the first mutation that you wrote:

```graphql
mutation {
  createTests(input: { _emptyInput: true }) {
    tests {
      test
    }
  }
}
```

Response:
`"Cannot return null for non-nullable field Test.test."`

Do you not receive the same response?

**Comment by michaeldaudiolo** [comment]
Hi, @mjfwebb. Sorry, you have to pass the argument through as true otherwise you get the same error
```
mutation {
  createTests(input: { _emptyInput: true }) {
    tests {
      test(value: true)
    }
  }
}
```

**Comment by neo4j-team-graphql** [comment]
We've been able to confirm this bug using the steps to reproduce that you provided - many thanks @michaeldaudiolo! :pray: We will now prioritise the bug and address it appropriately.

**Comment by MacondoExpress** [comment]
The fix for this issue is released on version 5.2.0, please be aware of another issue we found during the reproduction of this: https://github.com/neo4j/graphql/issues/4838 that is still to be fixed.
